Assets/Scripts/SceneLoader.cs

```diff
@@ -1,113 +1,126 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using System.IO;
 using UnityEngine;
 using UnityEngine.SceneManagement;
 using TMPro;
 using UnityEngine.UI;
+using System;
 
 public class SceneLoader : MonoBehaviour
 {
     public GameObject LevelSelector;
     public GameObject LevelContent;
-    private string starprogress = "";
-    // Start is called before the first frame update
-    void Start()
+    private string playerMinSaves = "0;-1;-1;-1;-1;-1;-1;-1;-1;-1";
+
+    // Stores max number of saves required per star per level
+    private List<List<int>> savesRequired = new List<List<int>>();
+
+    // Read information about max number of saves required per star per level
+    private void ParseStarInfo()
     {
+        string text = File.ReadAllText("./Assets/Resources/starinfo.txt");
+        string[] lines = text.Split('\n');
+
+        savesRequired = new List<List<int>>();
+        foreach (string line in lines)
+        {
+            if (line == "") break;
+            string[] info = line.Split(' '); // level number, 1 star, 2 star, 3 star
+            List<int> saves = new List<int> (new int[] { int.Parse(info[1]), int.Parse(info[2]), int.Parse(info[3]) } );
```

Comment:
Make savesRequired a list of SaveRequirements
Create class SaveRequirements with the info fields

Reply:
Used a local struct instead